extensions/2.0/Vendor/FB_ngon_encoding/README.md

```diff
@@ -0,0 +1,109 @@
+# FB_ngon_encoding
+
+## Contributors
+
+* Pär Winzell, Facebook, [zell@fb.com](mailto:zell@fb.com)
+* Michael Bunnell, Facebook
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+While glTF can only deliver a polygon mesh after it's been decomposed into triangles, there are cases where access to the source topology is still useful. One common such case is [Catmull-Clarke subdivision surfaces](https://en.wikipedia.org/wiki/Catmull%E2%80%93Clark_subdivision_surface), which work best with quadrilaterals.
+
+
+This extension contains **no additional data**:
+```
+  "meshes": [
+    {
+      "name": "Apple",
+      "primitives": [
+        {
+          "indices": 0,
+          "attributes": {
+            "POSITION": 1,
+            "NORMAL": 2,
+            "COLOR_0": 3,
+            "TEXCOORD_0": 4
+          },
+          "material": 0,
+          "mode": 4,
+          "extensions": {
+            "FB_ngon_encoding": {}
+          }
+        }
+      ]
+    }
+  ],
+```
+
+Rather, it marks the use of an implicit scheme where the **order of triangles and per-triangle vertex indices** holds all the information needed to reconstruct the original structure, and can be generated via a simple triangulation process:
+
+- For each polygon `p`, choose one identifying vertex `v(p)`,
+- Break the polygon into a fan of triangles, all anchored in that same `v(p)`,
+  - (It's obvious that this can be done for convex polygons, e.g. quads, but the true constraint is a bit less strict.)
```

Comment:
The constraint is a star-shaped polygon: https://en.wikipedia.org/wiki/Star-shaped_polygon

Reply:
The true constraint is more strict than "star-shaped", and it doesn't have a name of which I'm aware.  For example, ere are a couple of star-shaped polygons that don't work:
<img width="221" height="198" alt="image" src="https://github.com/user-attachments/assets/59b0972e-2a02-428e-a7f3-64893c1162ee" /> <img width="221" height="198" alt="image" src="https://github.com/user-attachments/assets/4a69c7ec-6cf6-4301-a2ce-70dc293cb02e" />

@devernay earlier said:
> Only non-convex (even star-shaped) ngons with n>=5 may pose an issue.

But I can't think of any pentagons with an issue: it looks to me like you need n>=6 (e.g. the pink non-convex star-shaped hexagon shown above) for there to be a problem.

Reply:
@donhatch Those work fine, if you add one more vertex in the middle for all triangles to share:

<img width="221" height="198" alt="485916232-59b0972e-2a02-428e-a7f3-64893c1162ee" src="https://github.com/user-attachments/assets/16a9333f-ffef-4763-b010-3fe92360ed55" /> <img width="221" height="198" alt="485915952-4a69c7ec-6cf6-4301-a2ce-70dc293cb02e" src="https://github.com/user-attachments/assets/6efe95b1-3113-4238-8538-05dda49bf9bb" />

Reply:
@aaronfranke true, but not relevant, unless I completely misunderstood your claim.  If I understood correctly, you claimed that the class of polygons for which the described scheme (which doesn't include adding a vertex) works is the star-shaped polygons.  I'm saying that claim is wrong, as shown by these two counterexamples (star-shaped polygons for which the scheme doesn't work).

Reply:
@donhatch Star-shaped doesn't just mean "shaped like a star", it is specific math terminology: https://en.wikipedia.org/wiki/Star-shaped_polygon

Reply:
Randomly stumbling over this: 

I asked a related question on Math.SE 12 years ago: https://math.stackexchange.com/q/856399 And even though this referred to general domains (and not polygons), I also hoped that it there was a description that referred to a _specific_ (given) point. Right now, it only says: "It is a shape for which _such a point exists_", and ... what are computer scientists supposed to do with such a claim...?

However, I think that one could describe the condition here analogously to the definition from Wikipedia. 

For Star-Shaped: 

> ... a polygon that contains a point from which the entire polygon boundary is visible.

For what is needed here

> ... a polygon that **has a vertex** from which the entire polygon boundary is visible.

Right?

Reply:
> For what is needed here
> 
> > ... a polygon that **has a vertex** from which the entire polygon boundary is visible.
> 
> Right?

That sounds right to me.  Unfortunately I don't know a good clear name for this concept and can't think of a good one.

Reply:
In G4MF, I called this `"polytopeCells"`, intending to imply that multiple simplex cells (ex: triangles) combine together to form a polytope (ex: polygon), as opposed to other non-cell (non-triangle) ways to encode polytopes (polygons). So perhaps some variant of `polygon_triangles`, `ngon_triangles`, `ngon_tris`, `polygon_tris`, etc. The "encoding" part of the extension name doesn't do much work, so to speak.